core/src/main/java/fr/sncf/osrd/conflicts/IncrementalRequirementEnvelopeAdapter.kt

```diff
@@ -16,17 +17,37 @@ class IncrementalRequirementEnvelopeAdapter(
         pathBeginOff: Offset<TravelledPath>,
         pathEndOff: Offset<TravelledPath>
     ): Double {
-        if (envelopeWithStops == null) return Double.POSITIVE_INFINITY
+        if (envelopeWithStops == null) {
+            return Double.POSITIVE_INFINITY
+        }
         val begin = pathBeginOff.distance.meters
         val end = pathEndOff.distance.meters
-        if (max(0.0, begin) >= min(envelopeWithStops.endPos, end))
+        if (max(0.0, begin) >= min(envelopeWithStops.endPos, end)) {
             return Double.POSITIVE_INFINITY // no overlap
+        }
         return envelopeWithStops.maxSpeedInRange(
             max(begin, 0.0),
             min(end, envelopeWithStops.endPos)
         )
     }
 
+    override fun departureFromStop(stopOffset: Offset<TravelledPath>): Double {
+        if (envelopeWithStops == null) {
+            return Double.POSITIVE_INFINITY
+        }
+        val endPos = envelopeWithStops.endPos
+        if (stopOffset.distance.meters > endPos) {
+            return Double.POSITIVE_INFINITY
+        }
+        // (stop offset + 1 millimeter) is the first position after stop, used to approximate/hack
+        // stop-departure time
```

Comment:
It's a bit of a shame to need this trick, it's the `envelopeWithStops.interpolateTotalTime()` call just below that iterates over the stops to include their durations. We could imagine an envelope method to get time at a stop location that lets us input if we want the time before or after the stop.

But that's not much of a problem considering the envelopes shouldn't stay around for very long. 

Edit, wait actually, 1mm is weird here. It's either not enough or not required altogether. When adding the stop duration in `EnvelopeStopWrapper.interpolateTotalTime`, we use `arePositionsEqual` which has a 1cm tolerance. It seems to skip the stop if we're within 1cm of the stop location.

It *seems* to work according to the tests, but there's a lot of red flags and *potential* bugs around there

Reply:
After discussion:
The current code of `interpolateTotalTime` currently returns the end-time of a stop.
Which means that this 1mm hack is useless (but doesn't change behavior as there is a 1cm epsilon).

The interface of `EnveloppeInterpolate` could be improved to let the choice to get stop-start or stop-end.
Adding this is reasonnably doable but raises a few questions: so not doing it in this PR, but removing the useless 1mm hack (and adding comments).

Reply:
Note that the doc on interpolate is currently misleading or wrong, but I'm not digging this for now.